File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java

class Solution {
    public List<List<Integer>> levelOrder(TreeNode r) {

  if (r == null)
      
      return new ArrayList<>();
    List<List<Integer>> ans = new ArrayList<>();
    Queue<TreeNode> a = new LinkedList<>(Arrays.asList(r));
        
    while (!a.isEmpty()) {
        
      List<Integer> currLevel = new ArrayList<>();
      for (int size = a.size(); size > 0; --size) {
        TreeNode node = a.poll();
        currLevel.add(node.val);
        if (node.left != null)
          a.offer(node.left);
        if (node.right != null)
          a.offer(node.right);
      }
      ans.add(currLevel);
    }
    return ans;
  }
}

File: GO-Leet/102. Binary Tree Level Order Traversal.go
package main

func levelOrder(root *TreeNode) [][]int {
	res := make([][]int, 0)

	var helper func(*TreeNode, int)
	helper = func(root *TreeNode, level int) {
		if root == nil {
			return
		}

		if level > len(res) {
			res = append(res, []int{root.Val})
		} else {
			res[level-1] = append(res[level-1], root.Val)
		}

		helper(root.Left, level+1)
		helper(root.Right, level+1)
	}

	helper(root, 1)

	return res
}
